v3/integration/lints/lint/lint.go

```diff
@@ -0,0 +1,139 @@
+package lint
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+type Lint interface {
+	Lint(tree *ast.File, file *File) *Result
+	CheckApplies(tree *ast.File, file *File) bool
+}
+
+type Result struct {
+	message       string
+	codeCitations []string
+	citations     []string
+}
+
+func NewResult(message string) *Result {
+	return &Result{message: message}
+
+}
+
+func (r *Result) AddCodeCitation(start, end token.Pos, file *File) *Result {
+	srcCode := make([]byte, end-start)
+	reader := strings.NewReader(file.Src)
+	reader.ReadAt(srcCode, int64(start))
+	lineno := file.LineOf(start)
+	citation := fmt.Sprintf("File %s, line %d\n\n%s\n\n", file.Path, lineno, string(srcCode))
```

Comment:
OK, so the documentation is bad enough that it might be a sign it's not worth going down this path, but at once point, GitHub used to [Officially Document the use of problem matchers](https://github.community/t/problemmatcher-documentation/16395/5). It's unclear if that's only for actions (e.g. [setup-go](https://github.com/actions/setup-go/blob/3b4dc6cbed1779f759b9c638cb83696acea809d1/matchers.json#L1)), but I thought I'd just mention it in terms of thinking about making sure this error message was maximally useful for that case :)

Reply:
I my mind, maximum sexiness is achieved here if we managed to inject some API credentials and webhook to the PR so that it could post directly the code in question.

Reply:
I don't think you need to inject creds in this case, I _think_ it would just work with the default token.